src/registry/domain/options-sanitiser.js

```diff
@@ -50,6 +48,7 @@ module.exports = function(input) {
     .map(s => s.toLowerCase());
 
   options.port = process.env.PORT || options.port;
+  options.timeout = options.timeout || 1000 * 60 * 2;
```

Comment:
This is the timeout for the http server, after which a request gets terminated with a "ECONNRESET".
When trying to upload a big file, I changed this 10m and worked fine. I opted for a 2m default (which is the node default) but left customisable by the user.